### DISCUSSION TITLE
jupyter extension?

### DISCUSSION THREAD

**Discussion by johanatan** [clavascript/clavascript#354] [Ideas]
Hi,

Would it be feasible to use squint to allow for ClojureScript in a Jupyter notebook? I know there is clojupyter but it's less than desirable due to the dependency on the JVM and JupyterLite being based on pyodide. Even that should be technically possible with a WebAssembly implementation of the JVM but it sounds rather painful compared to just a simple embedding (a la Lua). And I actually prefer ClojureScript to Clojure for several other reasons.

Thanks!

**Comment by borkdude** [comment]
I don't have any experience using Jupyter so some pointers would be nice. The clojupyter project looks kinda complicated.

**Comment by johanatan** [comment]
Yes, also another reason I’m trying to avoid it. 

I think the way to go here would be just some JS magic to tie squint or scittle into the deno repl. Since deno understands JS, we would just need to hook the text input for a REPL element and redirect it through squint or scittle before forwarding on to the original deno handler. 

The more involved approach would be a full blown kernel which is documented here: 
https://jupyter-client.readthedocs.io/en/stable/kernels.html

**Comment by johanatan** [comment]
Note that if you are doing the former, JupyterLite is probably easier / more straightforward (as I think it is fully client side). I used the following repo to get a deployment going on github pages to play around with:
https://github.com/jupyterlite/demo
